src/AlcoholTool.cpp

```diff
@@ -0,0 +1,123 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AlcoholTool.h"
+#include "unit.h"
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QFormLayout>
+#include <QSpacerItem>
+#include <math.h>
+#include <brewtarget.h>
+#include "BtLineEdit.h"
+
+AlcoholTool::AlcoholTool(QWidget* parent) : QDialog(parent)
+{
+   doLayout();
+   connect( pushButton_convert, SIGNAL(clicked()), this, SLOT(convert()) );
+}
+
+void AlcoholTool::doLayout()
+{
+   resize(279, 96);
+   QHBoxLayout* hLayout = new QHBoxLayout(this);
+      QFormLayout* formLayout = new QFormLayout();
+         label_og = new QLabel(this);
+         lineEdit_og = new BtDensityEdit(this);
+            lineEdit_og->setMinimumSize(QSize(80, 0));
+            lineEdit_og->setMaximumSize(QSize(80, 16777215));
+            lineEdit_og->setProperty("forcedUnit",QVariant(QStringLiteral("displaySG")));
+
+        label_fg = new QLabel(this);
+        lineEdit_fg = new BtDensityEdit(this);
+           lineEdit_fg->setMinimumSize(QSize(80, 0));
+           lineEdit_fg->setMaximumSize(QSize(80, 16777215));
+           lineEdit_fg->setProperty("forcedUnit",QVariant(QStringLiteral("displaySG")));
+
+         label_result = new QLabel(this);
+            label_result ->setObjectName(QStringLiteral("label_results"));
+            label_result ->setContextMenuPolicy(Qt::CustomContextMenu);
+
+         lineEdit_result = new QLineEdit(this);
+            lineEdit_result->setMinimumSize(QSize(80, 0));
+            lineEdit_result->setMaximumSize(QSize(80, 16777215));
+            lineEdit_result->setReadOnly(true);
+            lineEdit_result->setObjectName(QStringLiteral("lineEdit_result"));
+
+
+         formLayout->setWidget(0, QFormLayout::LabelRole, label_og);
+         formLayout->setWidget(0, QFormLayout::FieldRole, lineEdit_og);
+         formLayout->setWidget(1, QFormLayout::LabelRole, label_fg);
+         formLayout->setWidget(1, QFormLayout::FieldRole, lineEdit_fg);
+         formLayout->setWidget(2, QFormLayout::LabelRole, label_result);
+         formLayout->setWidget(2, QFormLayout::FieldRole, lineEdit_result);
+
+
+         formLayout->setFieldGrowthPolicy(QFormLayout::ExpandingFieldsGrow);
+         QVBoxLayout* vLayout = new QVBoxLayout();
+         QSpacerItem* verticalSpacer = new QSpacerItem(20, 40, QSizePolicy::Minimum, QSizePolicy::Expanding);
+         pushButton_convert = new QPushButton(this);
+            pushButton_convert->setAutoDefault(false);
+            pushButton_convert->setDefault(true);
+         QSpacerItem* verticalSpacer2 = new QSpacerItem(20, 40, QSizePolicy::Minimum, QSizePolicy::Expanding);
+         vLayout->addItem(verticalSpacer);
+         vLayout->addWidget(pushButton_convert);
+         vLayout->addItem(verticalSpacer2);
+
+         QSpacerItem* verticalSpacer3 = new QSpacerItem(20, 40, QSizePolicy::Minimum, QSizePolicy::Expanding);
+         vLayout->addItem(verticalSpacer3);
```

Comment:
This layout scheme is very different from what I am used to see and I think different from what this project normally uses. I would suggest using a code formatting tool, but I assume you have done this for readability purposes? I am not sure how other feel but this is harder for me to read.